cpp/src_prims/linalg/svd.cuh

```diff
@@ -232,32 +234,35 @@ void svdReconstruction(math_t *U, math_t *S, math_t *V, math_t *out, int n_rows,
  * @param allocator device allocator for temporary buffers during computation
  */
 template <typename math_t>
-bool evaluateSVDByL2Norm(math_t *A_d, math_t *U, math_t *S_vec, math_t *V,
-                         int n_rows, int n_cols, int k, math_t tol,
-                         cublasHandle_t cublasH, cudaStream_t stream,
-                         std::shared_ptr<deviceAllocator> allocator) {
+bool evaluateSVDByL2Norm(raft::handle_t &handle, math_t *A_d, math_t *U,
+                         math_t *S_vec, math_t *V, int n_rows, int n_cols,
+                         int k, math_t tol, cudaStream_t stream) {
+  cublasHandle_t cublasH = handle.get_cublas_handle();
+  std::shared_ptr<raft::mr::device::allocator> allocator =
+    handle.get_device_allocator();
+
   int m = n_rows, n = n_cols;
 
   // form product matrix
-  device_buffer<math_t> P_d(allocator, stream, m * n);
-  device_buffer<math_t> S_mat(allocator, stream, k * k);
+  raft::mr::device::buffer<math_t> P_d(allocator, stream, m * n);
+  raft::mr::device::buffer<math_t> S_mat(allocator, stream, k * k);
   CUDA_CHECK(cudaMemsetAsync(P_d.data(), 0, sizeof(math_t) * m * n, stream));
```

Comment:
Not for today, but in the future should we have a `buffer.assign(0, size)` method for this common operation?

Reply:
That would be a good first issue for some. Overloading operators like `[]` similar to `std::vector` and `thrust::device_vector` in my opinion